src/rgw/motr/gc/gc.cc

```diff
@@ -411,7 +410,7 @@ int MotrGC::list(std::vector<std::unordered_map<std::string, std::string>> &gc_e
       if (!marker.empty()) {
         keys[0] = marker;
       }
-      rc = store->next_query_by_name(iname, keys, vals, obj_tag_prefix);
+      int rc = store->next_query_by_name(iname, keys, vals, obj_tag_prefix);
       if (rc < 0) {
         ldout(cct, 0) <<__func__<<": ERROR: NEXT query failed. rc="
           << rc << dendl;
```

Comment:
For line #417 (i.e. "return rc"), we should not return here. We need to continue to next queue until all queues are over. We may put warning message saying there was some error (i.e dump rc value) when processing the given queue. When all queues are over/done,  the caller will use gc_entries. if it is empty, the GC list would be empty else it would be with some entries.

Reply:
In case of failure for any index, unless someone sees the logs or warning messages, they will never know that they are given half cooked information and entries from some indices are missing. Also, later we will be building REST interface where we will not have the option to show warning messages.
I think it's better to error out here.

Reply:
Thinking from another perspective, we need to handle the scaling down of the number of gc queues. We are not deleting the extra gc queues and the objects in them on scaling down.
On the first point I now slightly agree with Jeet, the user should provide the correct config for api's to work correctly.
On k8's setup for GC, anyway, we are exposing the REST API's which will share the config set for radosgw itself so it should not contradict the way it was on dev setup.

Reply:
@jjxsg : It's not about whether someone will see rgw logs or not. It's about putting the correct information in rgw log when something is failing, irrespective of whether someone would look into rgw logs are not. When we have information in rgw logs, one can see that there were some failure for some queues.

The reason I said we should not return from that line, because the next queue processing may be successful - we never know. I hope I'm understanding the logic there.

Reply:
Lets say there are 4 queues, 1,2,3,4. There is some issue with 3 (we will get rc < 0). If we don't error out, we will receive entries from 1,2 and 4 and since we got an output we may never suspect that the out missing entries from 3.